flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -217,12 +218,27 @@ private static HiveConf createHiveConf(
                     HadoopUtils.possibleHadoopConfPaths(
                             new org.apache.flink.configuration.Configuration())) {
                 hadoopConf = getHadoopConfiguration(possibleHadoopConfPath);
-                if (hadoopConf != null) {
+                if (hadoopConf != null && hadoopConf.size() != 0) {
                     break;
                 }
             }
         } else {
             hadoopConf = getHadoopConfiguration(hadoopConfDir);
+            if (hadoopConf == null) {
+                throw new CatalogException(
+                        "Failed to load the hadoop conf from specified path:" + hadoopConfDir,
+                        new FileNotFoundException("Path " + hadoopConfDir + " does not exist."));
+            }
+            if (hadoopConf.size() == 0) {
```

Comment:
Same here.

Reply:
(1)The method `getHadoopConfiguration ` returns  `hadoopConfiguration` that is not null when hadoop path exists but no conf files exist.
(2)The method `getHadoopConfiguration ` returns  `hadoopConfiguration` that is null when hadoop path does not exist.
if hadoopConf == null , hadoop does not exist; 
if hadoopConf != null but hadoopConf.size == 0 , hadoop path exists but no conf files.